docs/proposals/platform-extension-implementation-extension-operators.md

```diff
@@ -0,0 +1,86 @@
+# Platform Extension Implementation: Extension Operators
+
+**Author**: Moritz Bracht
+
+**Status**: Draft proposal.
+
+## Goals
+
+This proposal explores an operator based approach for KKP extensions.
+
+## Motivation and Background
+
+An implementation of platform-extension based on Helm charts provides flexible
+yet simplistic way to manage the lifecycle of platform-extensions.
+
+In case of more complex extensions that can not simply be installed with Helm
+([like KubeVirt](https://kubevirt.io/user-guide/operations/installation/), the
+extensions could be managed by their own operators.
+
+To cover simple Helm based extensions either a Helm operator could be installed
+or simple operators could be [generated directly from Helm
+charts](https://sdk.operatorframework.io/docs/building-operators/helm/tutorial/).
+
+[Check this for requirements](./platform-extensions.md#requirements-for-implementations)
+
+## Implementation
+
+Every extension is managed by its own operator. These operators are being
+managed by [OLM](https://olm.operatorframework.io/).
+
+### Main Concept Points
+
+* Extensions provided by KKP devs
+  * Extension operators are bundled and provided via OLM catalog
+* Extensions provided by users
+  * Users can create and bundle their own extension operators and extend the
+	catalog
+* Life cycle management for extensions comes with OLM
```

Comment:
How transparent is OLM to integrate into KKP? Is it always an individual tool or does it provide an API we can address by KKP?

Reply:
As OLM is a direct extension of Kubernetes, its resources are directly available. As of KKP integration, that part of the proposal is still pretty much open. I would see the Extension Catalog Controller and the Extension Controller as a part of the KKP controller manager. They would reconcile KKP Extension related CRs and generate CRs reconciled by the OLM operators.

This sounds kind of layered and indirected, but maybe can result in a clean interface to benefit from OLM's toolkit.